src/net_processing.cpp

```diff
@@ -4057,7 +4132,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
             // Erase this entry from tx_process_time (it may be added back for
             // processing at a later time, see below)
             tx_process_time.erase(tx_process_time.begin());
-            CInv inv(MSG_TX | GetFetchFlags(pto), txid);
+            CInv inv(state.m_wtxid_relay ? MSG_WTX : (MSG_TX | GetFetchFlags(pto)), txid);
```

Comment:
Seeing a build warning here:
```
net_processing.cpp: In member function ‘virtual bool PeerLogicValidation::SendMessages(CNode*)’:
net_processing.cpp:4135:42: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
             CInv inv(state.m_wtxid_relay ? MSG_WTX : (MSG_TX | GetFetchFlags(pto)), txid);
                      ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Yeah I saw that too, but I think the code is correct?

Reply:
I think it's treating `MSG_WTX` as `enum GetDataMsg` and `MSG_TX | GetFetchFlags()` as `uint32_t` (match `GetFetchFlags` return type). Changing `GetFetchFlags()` to return `GetDataMsg` (and casting its result correspondingly), and making it `enum GetDataMsg : uint32_t` makes the warning go away. Actually, just saying `enum GetDataMsg : uint32_t` makes the warning go away too by the looks.

Reply:
Updated with the `enum GetDataMsg : uint32_t` fix, thanks.